merlin/systems/dag/dictarray.py

```diff
@@ -13,44 +13,149 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from enum import Enum
 from typing import Dict, Optional
 
 import numpy as np
 
-from merlin.core.protocols import SeriesLike, Transformable
+from merlin.core.dispatch import HAS_GPU
+from merlin.core.protocols import SeriesLike
+
+if HAS_GPU:
+    import cupy
+else:
+    cupy = None
+
+
+class Device(Enum):
+    CPU = 0
+    GPU = 1
 
 
 class Column(SeriesLike):
     """
-    A simple wrapper around an array of values
+    A simple wrapper around an array of values. This has an API that's just similar enough to
+    Pandas and cuDF series to be relatively interchangeable from the perspective of the Merlin DAG,
+    but no more. As more methods get added to this class, it gets closer and closer to actually
+    *being* a Pandas/cuDF Series (at which point there's no advantage to using this.) So: keep
+    this class as small as possible.
     """
 
-    def __init__(self, values):
+    def __init__(self, values, row_lengths=None):
         super().__init__()
 
         if isinstance(values, Column):
             raise TypeError("doubly nested columns")
 
         self.values = values
+        self.row_lengths = row_lengths
         self.dtype = values.dtype
 
+        if isinstance(values, np.ndarray):
```

Comment:
We hadn't intended to tackle the scope of moving data between host and device memory here, but it cropped up in some of the Feast tests so we went ahead and did it.